packages/@aws-cdk/aws-dynamodb/test/test.dynamodb.ts

```diff
@@ -229,8 +229,791 @@ export = {
+                            Effect: 'Allow', Resource: '*' } ],
+                       Version: '2012-10-17' },
+                    PolicyName: 'MyTableReadAutoScalingRoleDefaultPolicyF6A1975F',
+                    Roles: [ { Ref: 'MyTableReadAutoScalingRoleFEE68E49' } ] } },
+              MyTableReadCapacityScalableTarget72B0B3BF:
+               { Type: 'AWS::ApplicationAutoScaling::ScalableTarget',
+                 Properties:
+                  { MaxCapacity: 500,
+                    MinCapacity: 50,
+                    ResourceId:
+                     { 'Fn::Join': [ '', [ 'table/', { Ref: 'MyTable794EDED1' } ] ] },
+                    RoleARN:
+                     { 'Fn::GetAtt': [ 'MyTableReadAutoScalingRoleFEE68E49', 'Arn' ] },
+                    ScalableDimension: 'dynamodb:table:ReadCapacityUnits',
+                    ServiceNamespace: 'dynamodb' } },
+              MyTableReadCapacityScalingPolicyCC18E396:
+               { Type: 'AWS::ApplicationAutoScaling::ScalingPolicy',
+                 Properties:
+                  { PolicyName: 'MyAwesomePolicyName',
+                    PolicyType: 'TargetTrackingScaling',
+                    ScalingTargetId: { Ref: 'MyTableReadCapacityScalableTarget72B0B3BF' },
+                    TargetTrackingScalingPolicyConfiguration:
+                     { PredefinedMetricSpecification: { PredefinedMetricType: 'DynamoDBReadCapacityUtilization' },
+                       ScaleInCooldown: 80,
+                       ScaleOutCooldown: 60,
+                       TargetValue: 75 } } } } });
+
+        test.done();
+    },
+
+    'when specifying Read Auto Scaling via constructor and attempting to addReadAutoScaling - throws Error'(test: Test) {
+        const app = new TestApp();
+        const table = new Table(app.stack, 'MyTable', {
+            tableName: 'MyTable',
+            readCapacity: 42,
+            writeCapacity: 1337,
+            readAutoScaling: {
+                minimumCapacity: 50,
+                maximumCapacity: 500,
+                scalingTargetValue: 75.0,
+                scaleInCooldown: 80,
+                scaleOutCooldown: 60,
+                scalingPolicyName: 'MyAwesomePolicyName'
+            }
+        });
+        table.addPartitionKey('partitionKey', KeyAttributeType.String);
+        table.addSortKey('sortKey', KeyAttributeType.Binary);
+        test.throws(() => table.addReadAutoScaling({
+            minimumCapacity: 500,
+            maximumCapacity: 5000,
+            scalingTargetValue: 25.0,
+            scaleInCooldown: 40,
+            scaleOutCooldown: 20,
+            scalingPolicyName: 'MySecondAwesomePolicyName'
+        }), Error, "Read Auto Scaling already defined for Table");
+
+        test.done();
+    },
+
+    'when specifying Read Auto Scaling without scalingPolicyName'(test: Test) {
+        const app = new TestApp();
+        const table = new Table(app.stack, 'MyTable', {
+            tableName: 'MyTable',
```

Comment:
Test this without specifying a `tableName` (so it's a {Ref})

Reply:
done